plugin/src/main/java/com/denizenscript/denizen/events/server/ListPingScriptEvent.java

```diff
@@ -52,23 +51,20 @@ public class ListPingScriptEvent extends BukkitScriptEvent implements Listener {
 
     public ListPingScriptEvent() {
         registerCouldMatcher("server list ping");
-    }
-
-    public ServerListPingEvent event;
-
-    // Despite the 'cached' class name, there's no actual internal cache.
-    public static HashMap<String, CachedServerIcon> iconCache = new HashMap<>();
-
-    public void setMotd(String text) {
-        event.setMotd(text);
-    }
-
-    @Override
-    public boolean applyDetermination(ScriptPath path, ObjectTag determinationObj) {
-        String determination = determinationObj.toString();
-        String determineLow = CoreUtilities.toLowerCase(determination);
-        if (determineLow.startsWith("icon:")) {
-            String iconFile = determination.substring("icon:".length());
+        this.<ListPingScriptEvent, ListTag>registerOptionalDetermination(null, ListTag.class, (evt, context, list) -> {
+            if (ArgumentHelper.matchesInteger(list.get(0))) {

```

Comment:
a proper modernization should probably add dedicated `determine MOTD:<text>` and `determine max_players:<#>` options